pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -93,45 +93,42 @@ type VirtualHostWrapper struct {
 	Routes []*route.Route
 }
 
-// BuildSidecarVirtualHostWrapper creates virtual hosts from
-// the given set of virtual Services and a list of Services from the
-// service registry. Services are indexed by FQDN hostnames.
+// BuildSidecarVirtualHostWrapper creates virtual hosts from the given set of virtual Services
+// and a list of Services from the service registry. Services are indexed by FQDN hostnames.
 // The list of Services is also passed to allow maintaining consistent ordering.
 func BuildSidecarVirtualHostWrapper(routeCache *Cache, node *model.Proxy, push *model.PushContext, serviceRegistry map[host.Name]*model.Service,
 	virtualServices []config.Config, listenPort int,
 ) []VirtualHostWrapper {
 	out := make([]VirtualHostWrapper, 0)
 
-	// dependentDestinationRules includes all the destinationrules referenced by the virtualservices, which have consistent hash policy.
+	// dependentDestinationRules includes all the destinationrules referenced by
+	// the virtualservices, which have consistent hash policy.
 	dependentDestinationRules := []*model.ConsolidatedDestRule{}
 
-	// translate all virtual service configs into virtual hosts
+	// First build virtual host wrappers for services that have virtual services.
 	for _, virtualService := range virtualServices {
-		hashByDestination, destinationRules := hashForVirtualService(push, node, virtualService, serviceRegistry)
+		hashByDestination, destinationRules := hashForVirtualService(push, node, virtualService)
 		dependentDestinationRules = append(dependentDestinationRules, destinationRules...)
 		wrappers := buildSidecarVirtualHostsForVirtualService(node, virtualService, serviceRegistry, hashByDestination, listenPort, push.Mesh)
 		out = append(out, wrappers...)
 	}
 
-	// compute Services missing virtual service configs
+	// Now exclude the services that have virtual services.
 	for _, wrapper := range out {
 		for _, service := range wrapper.Services {
 			delete(serviceRegistry, service.Hostname)
 		}
 	}
 
-	hashByService := map[host.Name]map[int]*networking.LoadBalancerSettings_ConsistentHashLB{}
 	for _, svc := range serviceRegistry {
 		for _, port := range svc.Ports {
 			if port.Protocol.IsHTTP() || util.IsProtocolSniffingEnabledForPort(port) {
-				hash, destinationRule := getHashForService(node, push, svc, port)
+				hash, destinationRule := hashForService(node, push, svc, port)
```

Comment:
@hzxuzhonghu I know you preferred previous name. But changed to be consistent with `hashForVirtualService`